OpenSim/Simulation/SimulationUtilities.cpp

```diff
@@ -175,3 +175,32 @@ void OpenSim::updatePre40KinematicsFilesFor40MotionType(const Model& model,
         updatedMotion->print(outFilePath);
     }
 }
+
+void OpenSim::updateConnecteesBySearch(Model& model)
+{
+    for (auto& comp : model.updComponentList()) {
+        const auto socketNames = comp.getSocketNames();
+        for (int i = 0; i < socketNames.size(); ++i) {
+            auto& socket = comp.updSocket(socketNames[i]);
+            try {
+                socket.finalizeConnection(model);
+            } catch (const ComponentNotFoundOnSpecifiedPath&) {
+                const ComponentPath path(socket.getConnecteePath());
+                if (path.getNumPathLevels() >= 1) { 
+                    const Component* found =
+                        model.findComponent(path.getComponentName());
+                    if (found) {
+                        socket.connect(*found);
+                        socket.finalizeConnection(model);
+                    }
```

Comment:
else, report that this Socket needs updating but couldn't be done automatically?

Reply:
Done.